### PR TITLE
ReactiveBuilder for RxJava

### PR DESCRIPTION
**Pull request by wglanzer**
This PullRequest contains the first version of the ObservableBuilder-Implementation.
Maybe we should additionaly increase the version - 0.5,0?

### CODE REVIEW

File: propertly.reactive/src/main/java/de/adito/propertly/reactive/api/PropertlyReactiveException.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package de.adito.propertly.reactive.api;
+
+/**
+ * Exception for onError-Processing inside Propertly Observables
+ *
+ * @author w.glanzer, 01.11.2018
+ */
+@SuppressWarnings("unused")
+public class PropertlyReactiveException extends Exception
```
Comment (jboesl) [inline propertly.reactive/src/main/java/de/adito/propertly/reactive/api/PropertlyReactiveException.java]: This exception is unused. Maybe remove it?

File: propertly.reactive/src/main/java/de/adito/propertly/reactive/api/builder/IBuilderObservable.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import java.util.Optional;
+
+/**
+ * Common Interface for all Propertly-Observables constructed by PropertlyObservableBuilder
+ *
+ * @see PropertlyObservableBuilder
+ * @author w.glanzer, 09.11.2018
+ */
+public interface IBuilderObservable<V>
```
Comment (jboesl) [inline propertly.reactive/src/main/java/de/adito/propertly/reactive/api/builder/IBuilderObservable.java]: I think all 'IBuilder...' interfaces should be 'IProperty...Observable' interfaces. It is started with a build but the object you receive is no longer a builder.

File: propertly.reactive/src/main/java/de/adito/propertly/reactive/api/builder/IBuilderPropertyListObservable.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+   * The resulting Observable is not distincted automatically. <br>
+   * All properties are observed, so the Observable will get notified if any of the
+   * containing properties changes.
+   *
+   * @param pIncludeNullValues <tt>true</tt>, if the inner list should contain NULL-Values, if a property has <tt>null</tt> as a value
+   * @return the Observable, not <tt>null</tt>
+   * @see IBuilderPropertyListObservable#emitDistinctedValues(boolean)
+   */
+  @NotNull
+  Observable<Optional<List<T>>> emitValues(boolean pIncludeNullValues);
```
Comment (jboesl) [inline propertly.reactive/src/main/java/de/adito/propertly/reactive/api/builder/IBuilderPropertyListObservable.java]: Do you need 'Optional' here? You could return an empty list, too.

Comment (wglanzer) [reply propertly.reactive/src/main/java/de/adito/propertly/reactive/api/builder/IBuilderPropertyListObservable.java]: Optional.empty = Chain is invalid / Value can not be determined
Collections.emptyList = No Values, but valid chain

**Pull request merged**